app/controllers/annotations_controller.rb

```diff
@@ -163,18 +163,20 @@ def create
       handleRequiredListMultipleOn
       @annotationOut['canvas'] = setMultipleCanvas
     end
-
+   
     p "in CreateAnno: @annotationOut['canvas'] = #{@annotationOut['canvas']}"
     p "in CreateAnno: about to setMap: @annotationIn['within'] = #{@annotationIn['within']}"
     ListAnnotationsMap.setMap @annotationIn['within'], @annotation_id
     create_annotation_acls_via_parent_lists @annotation_id
     @annotation = Annotation.new(@annotationOut)
+    
+    return unless check_anno_auth(request, @annotation, "There was an error creating the annotation").nil?
```

Comment:
Returning nil for success bother me a little. I'd prefer a checker method return a boolean.
Something like this may be clearer:
```
unless anno_authorized(request, @annotation)
  return render_failure('User not authorized to create the annotation')
end
```

Reply:
Good idea, I refactored into two functions.